platform/darwin/src/MGLGeometry_Private.h

```diff
@@ -13,6 +13,18 @@
 CGRect MGLExtendRect(CGRect rect, CGPoint point);
 
 NS_INLINE mbgl::LatLng MGLLatLngFromLocationCoordinate2D(CLLocationCoordinate2D coordinate) {
+    if (std::isnan(coordinate.latitude)) {
+        [NSException raise:NSInvalidArgumentException format:@"latitude must not be NaN"];
+    }
+    if (std::isnan(coordinate.longitude)) {
+        [NSException raise:NSInvalidArgumentException format:@"longitude must not be NaN"];
+    }
+    if (std::abs(coordinate.latitude) > 90.0) {
```

Comment:
My suggestion was to move these checks to individual call sites, or to add a “checked” version of this function that wraps `MGLLatLngFromLocationCoordinate2D()`, adding these checks.

I reviewed the existing call sites to `MGLLatLngFromLocationCoordinate2D()` and tested similar functionality in MapKit using invalid coordinates. Based on MapKit behavior, having the exceptions where they are is OK, although we’ll want to add identical preflight checks in methods like `-[MGLMapView convertCoordinate:toPointToView:]` that return things like `CGPointMake(NAN, NAN)`.

Mapbox call site | MapKit equivalent | MapKit behavior | Mapbox expected behavior
----|----|----|----
`-[MGLMapView handlePinchGesture:]` | — | Constrained | `NSAssert` (but `NSException` is fine)
`-[MGLMapView _setCenterCoordinate:edgePadding:zoomLevel:direction:duration:animationTimingFunction:completionHandler:]` | `-[MKMapView setRegion:animated:]` | `NSException` | `NSException`
`-[MGLMapView cameraOptionsObjectForAnimatingToCamera:edgePadding:]` | `-[MKMapView setCamera:animated:]` | `NSException` | `NSException`
`-[MGLMapView convertCoordinate:toPointToView:]` | `-[MKMapView convertCoordinate:toPointToView:]` | `CGPoint(NaN, NaN)` | `CGPoint(NaN, NaN)`
`-[MGLMapView showAnnotations:edgePadding:animated:]` | `-[MKMapView showAnnotations:animated:]` | Moves to invalid part of map, in ocean | `NSException` is OK
`-[MGLMapView directionByFollowingWithCourse]` | — | — | `NSException`